core/quivr_core/rag/entities/config.py

```diff
@@ -0,0 +1,452 @@
+                    self.conditions[index] = END
+                elif value == SpecialEdges.start:
+                    self.conditions[index] = START
+
+
+class NodeConfig(QuivrBaseConfig):
+    name: str
+    edges: List[str] | None = None
+    conditional_edge: ConditionalEdgeConfig | None = None
+    tools: List[Dict[str, Any]] | None = None
+    instantiated_tools: List[BaseTool | Type] | None = None
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._instantiate_tools()
+        self.resolve_special_edges_in_name_and_edges()
+
+    def resolve_special_edges_in_name_and_edges(self):
+        """Replace SpecialEdges enum values in name and edges with corresponding langgraph values."""
+        if self.name == SpecialEdges.start:
+            self.name = START
+        elif self.name == SpecialEdges.end:
+            self.name = END
+
+        if self.edges:
+            for i, edge in enumerate(self.edges):
+                if edge == SpecialEdges.start:
+                    self.edges[i] = START
+                elif edge == SpecialEdges.end:
+                    self.edges[i] = END
+
+    def _instantiate_tools(self):
+        """Instantiate tools based on the configuration."""
+        if self.tools:
+            self.instantiated_tools = [
+                LLMToolFactory.create_tool(tool_config.pop("name"), tool_config)
+                for tool_config in self.tools
+            ]
+
+
+class DefaultWorkflow(str, Enum):
+    RAG = "rag"
+
+    @property
+    def nodes(self) -> List[NodeConfig]:
+        # Mapping of workflow types to their default node configurations
+        workflows = {
+            self.RAG: [
+                NodeConfig(name=START, edges=["filter_history"]),
+                NodeConfig(name="filter_history", edges=["rewrite"]),
+                NodeConfig(name="rewrite", edges=["retrieve"]),
+                NodeConfig(name="retrieve", edges=["generate_rag"]),
+                NodeConfig(name="generate_rag", edges=[END]),
+            ]
+        }
+        return workflows[self]
+
+
+class WorkflowConfig(QuivrBaseConfig):
+    name: str | None = None
+    nodes: List[NodeConfig] = []
+    available_tools: List[str] | None = None
+    validated_tools: List[BaseTool | Type] = []
+    activated_tools: List[BaseTool | Type] = []
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.check_first_node_is_start()
+        self.validate_available_tools()
+
+    def check_first_node_is_start(self):
+        if self.nodes and self.nodes[0].name != START:
+            raise ValueError(f"The first node should be a {SpecialEdges.start} node")
+
+    def get_node_tools(self, node_name: str) -> List[Any]:
+        """Get tools for a specific node."""
+        for node in self.nodes:
+            if node.name == node_name and node.instantiated_tools:
+                return node.instantiated_tools
+        return []
+
+    def validate_available_tools(self):
```

Comment:
We could leverage pydantic validators here: 
https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators

This means we can remove the validate in __init__

Reply:
indeed, but we first we should remove any dependency on pydantic V1...